server/db/driver/pg/orders.go

```diff
@@ -7,105 +7,55 @@ import (
 	"context"
 	"database/sql"
 	"database/sql/driver"
-	"encoding/hex"
 	"fmt"
-	"strconv"
-	"strings"
 
 	"decred.org/dcrdex/dex"
 	"decred.org/dcrdex/dex/order"
 	"decred.org/dcrdex/server/account"
 	"decred.org/dcrdex/server/db"
 	"decred.org/dcrdex/server/db/driver/pg/internal"
-	"github.com/lib/pq"
 )
 
-// utxo implements order.Outpoint
-type utxo struct {
-	txHash []byte
-	vout   uint32
-}
-
-func (u *utxo) TxHash() []byte { return u.txHash }
-func (u *utxo) Vout() uint32   { return u.vout }
-
-func newUtxo(txid string, vout uint32) *utxo {
-	hash, err := hex.DecodeString(txid)
-	if err != nil {
-		panic(err)
-	}
-	return &utxo{hash, vout}
-}
+// Wrap the CoinID slice to implement custom Scanner and Valuer.
+type dbCoins []order.CoinID
 
-func newUtxoFromOutpoint(outpoint string) (*utxo, error) {
-	outParts := strings.Split(outpoint, ":")
-	if len(outParts) != 2 {
-		return nil, fmt.Errorf("invalid outpoint %s", outpoint)
-	}
-	hash, err := hex.DecodeString(outParts[0])
-	if err != nil {
-		panic(err)
+// Value implements the sql/driver.Valuer interface.
+func (coins dbCoins) Value() (driver.Value, error) {
+	if len(coins) == 0 {
+		return []byte{}, nil
 	}
-	vout, err := strconv.ParseUint(outParts[1], 10, 32)
-	if err != nil {
-		return nil, fmt.Errorf("invalid vout %s: %v", outParts[1], err)
+	b := make([]byte, 0, len(coins)*(len(coins[0])+1))
+	for _, coin := range coins {
+		b = append(b, byte(len(coin)))
+		b = append(b, coin...)
 	}
-	return &utxo{hash, uint32(vout)}, nil
-}
-
-// Value implements the sql/driver.Valuer interface.
-func (u utxo) Value() (driver.Value, error) {
-	// UTXOs are stored as an array of strings like ["txid0:vout0", ...] despite
-	// this being less space efficient than a BYTEA because it significantly
-	// simplifies debugging.
-	return order.OutpointString(&u), nil
+	return b, nil
 }
 
 // Scan implements the sql.Scanner interface.
-func (u *utxo) Scan(src interface{}) error {
-	var opStr string
-	switch st := src.(type) {
-	case string:
-		opStr = st
-	case []byte: // via pq.GenericArray.Scan
-		opStr = string(st)
-	default:
-		return fmt.Errorf("cannot convert %T to utxo", src)
+func (coins *dbCoins) Scan(src interface{}) error {
+	b := src.([]byte)
+	if len(b) == 0 {
+		return nil
```

Comment:
Before `Scan` the caller will almost certainly be starting with an empty `dbCoins` slice, but I wonder if we should explicitly `*coins = dbCoins{}` before returning here to be safe (in case the caller is doing something bad like reusing a scan variable).